lgc/tool/lgc/lgc.cpp

```diff
@@ -0,0 +1,259 @@
+      "                             replaced to match the output type\n"
+      "  -mcpu=<cpu-name>         - Target a specific GPU e.g. 'gfx900'\n"
+      "  -filetype=<value>        - Choose an output file type:\n"
+      "   =asm                    -   Emit an assembly ('.s') file (default)\n"
+      "   =obj                    -   Emit an ELF ('.elf') file\n"
+      "  -emit-llvm               - Emit LLVM IR assembly before running back-end\n"
+      "  -pal-abi-version=<uint>  - PAL pipeline ABI version to compile for (default\n"
+      "                             latest known)\n"
+      "  -verify-ir               - Verify IR after each pass\n"
+      );
+
+  // Find the -mcpu option and get its value.
+  auto opts = cl::getRegisteredOptions();
+  auto mcpu = opts.find("mcpu");
+  assert(mcpu != opts.end());
+  auto *mcpuOpt = reinterpret_cast<cl::opt<std::string> *>(mcpu->second);
+  StringRef gpuName = *mcpuOpt;
+  if (gpuName == "") {
+    gpuName = "gfx802";
+  }
+
+  // Create the LgcContext.
+  std::unique_ptr<LgcContext> lgcContext(LgcContext::Create(context, gpuName, PalAbiVersion));
+  if (!lgcContext) {
+    errs() << progName << ": GPU type '" << gpuName << "' not recognized\n";
+    return 1;
+  }
+
+  for (auto inFileName : InFiles) {
+    // Read the input file. getFileOrSTDIN handles the case of inFileName being "-".
+    ErrorOr<std::unique_ptr<MemoryBuffer>> fileOrErr = MemoryBuffer::getFileOrSTDIN(inFileName);
+    if (std::error_code errorCode = fileOrErr.getError()) {
+      auto error = SMDiagnostic(inFileName, SourceMgr::DK_Error, "Could not open input file: " + errorCode.message());
+      error.print(progName, errs());
+      errs() << "\n";
+      return 1;
+    }
+    StringRef bufferName = fileOrErr.get()->getMemBufferRef().getBufferIdentifier();
+
+    // Split the input into multiple LLVM IR modules. We assume that a new module starts with
+    // a "target" line to set the datalayout or triple, but not until after we have seen at least
+    // one line starting with '!' (metadata declaration) in the previous module.
+    SmallVector<StringRef, 4> separatedAsms;
+    StringRef remaining = fileOrErr.get()->getMemBufferRef().getBuffer();
+    separatedAsms.push_back(remaining);
+    bool hadMetadata = false;
+    for (;;) {
+      auto notSpacePos = remaining.find_first_not_of(" \t\n");
+      if (notSpacePos != StringRef::npos) {
+        if (remaining[notSpacePos] == '!')
+          hadMetadata = true;
+        else if (hadMetadata && remaining.slice(notSpacePos, StringRef::npos).startswith("target")) {
+          // End the current split module and go on to the next one.
+          separatedAsms.back() = separatedAsms.back().slice(0, remaining.data() - separatedAsms.back().data());
+          separatedAsms.push_back(remaining);
+          hadMetadata = false;
+        }
+      }
+      auto nlPos = remaining.find_first_of('\n');
```

Comment:
nlPos -> newLinePos

Reply:
I prefer nlPos as NL is a well-known abbreviation for newline.